### PR TITLE
LiteCore updates

### PR DESCRIPTION
**Pull request by Sandychuang8**
C4DatabaseConfig2 replaces C4DatabaseConfig
Used new c4db_xxNamed methods
c4db_getDoc replaces c4doc_get 
c4doc_getRevisionBody replaces RawDoc->selectedRev.body
MATCH() replaces MATCH
Tests update with above updates

### CODE REVIEW

**Comment by borrrden** [comment]
What should I review here?

**Comment by Sandychuang8** [comment]
See if you spot any error I made with using the latest LiteCore API.

**Comment by borrrden** [comment]
That's pretty huge and lacks a description with 26 files changed.  I will look it over but it will probably take a while.

**Comment by Sandychuang8** [comment]
Sorry. Just added a short summarization to included what changes are made in this PR in the PR description.

File: src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs [hunk 1]
```diff
@@ -64,6 +64,20 @@ public string Directory
...
+        /// <remarks>
+        /// If the enableVersionVector is set to true, the database will use version vector instead of
+        /// using revision tree.When enabling version vector on an existing database, the database
+        /// will be upgraded to use the revision tree while the database is opened.
+        /// NOTE:
+        /// 1. The database that uses version vector cannot be downgraded back to use revision tree.
+        /// 2. The current version of Sync Gateway doesn't support version vector so the syncronization
+        /// with Sync Gateway will not be working.
+        /// </remarks>
+        internal bool EnableVersionVector => false;
```
Comment (borrrden) [inline src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:79]: Is this ok to be internal?  Is QE expecting to use this?

**Review by borrrden — APPROVED** [review]
The only comment I have is not enough to deny the PR

File: src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs (hunk 1 above)
Comment (Sandychuang8) [reply src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:79]: This will be enabled once VV is ready for testing.

**Pull request merged**